libr/util/sys.c

```diff
@@ -363,26 +363,34 @@ R_API int r_sys_crash_handler(const char *cmd) {
 
 	free (crash_handler_cmd);
 	crash_handler_cmd = strdup (cmd);
-	sigact.sa_handler = signal_handler;
-	sigemptyset (&sigact.sa_mask);
-	sigact.sa_flags = 0;
-	sigaddset (&sigact.sa_mask, SIGINT);
-	sigaddset (&sigact.sa_mask, SIGSEGV);
-	sigaddset (&sigact.sa_mask, SIGBUS);
-	sigaddset (&sigact.sa_mask, SIGQUIT);
-	sigaddset (&sigact.sa_mask, SIGHUP);
+	sigemptyset (&mask);
+	sigaddset (&mask, SIGINT);
+	sigaddset (&mask, SIGSEGV);
+	sigaddset (&mask, SIGBUS);
+	sigaddset (&mask, SIGQUIT);
+	sigaddset (&mask, SIGHUP);
 
-	sigaction (SIGINT, &sigact, (struct sigaction *)NULL);
-	sigaction (SIGSEGV, &sigact, (struct sigaction *)NULL);
-	sigaction (SIGBUS, &sigact, (struct sigaction *)NULL);
-	sigaction (SIGQUIT, &sigact, (struct sigaction *)NULL);
-	sigaction (SIGHUP, &sigact, (struct sigaction *)NULL);
+	r_sys_sigaction (SIGINT, signal_handler, &mask, 0);
+	r_sys_sigaction (SIGSEGV, signal_handler, &mask, 0);
+	r_sys_sigaction (SIGBUS, signal_handler, &mask, 0);
+	r_sys_sigaction (SIGQUIT, signal_handler, &mask, 0);
+	r_sys_sigaction (SIGHUP, signal_handler, &mask, 0);
 	return true;
 #else
 	return false;
 #endif
 }
 
+#if __UNIX__
+R_API int r_sys_sigaction(int sig, void(*handler) (int), sigset_t *mask, int flags) {
```

Comment:
add #if USE_SIGNAL to just call `signal (2)` instead of all that sigaction thing if not supported by the target platform. and `#define USE_SIGNAL 0` at the begining of the file